tests/functional/codegen/features/test_transient.py

```diff
@@ -55,3 +47,353 @@ def setter(k: address, v: uint256):
 
     assert "TLOAD" in t
     assert "TSTORE" in t
+
+
+@pytest.mark.requires_evm_version("cancun")
+@pytest.mark.parametrize(
+    "typ,value,zero",
+    [
+        ("uint256", 42, 0),
+        ("int256", -(2**200), 0),
+        ("int128", -(2**126), 0),
+        ("address", "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE", None),
+        ("bytes32", b"deadbeef" * 4, b"\x00" * 32),
+        ("bool", True, False),
+        ("String[10]", "Vyper hiss", ""),
+        ("Bytes[10]", b"Vyper hiss", b""),
+    ],
+)
+def test_value_storage_retrieval(typ, value, zero, get_contract):
+    code = f"""
+bar: public(transient({typ}))
+
+@external
+def foo(a: {typ}) -> {typ}:
+    self.bar = a
+    return self.bar
+    """
+
+    c = get_contract(code)
+    assert c.bar() == zero
+    assert c.foo(value) == value
+    assert c.bar() == zero
+    assert c.foo(value) == value
+    assert c.bar() == zero
+
+
+@pytest.mark.requires_evm_version("cancun")
+@pytest.mark.parametrize("val", [0, 1, 2**256 - 1])
+def test_usage_in_constructor(get_contract, val):
+    code = """
+A: transient(uint256)
+a: public(uint256)
+
+
+@deploy
+def __init__(_a: uint256):
+    self.A = _a
+    self.a = self.A
+
+
+@external
+@view
+def a1() -> uint256:
+    return self.A
+    """
+
+    c = get_contract(code, val)
+    assert c.a() == val
+    assert c.a1() == 0
+
+
+@pytest.mark.requires_evm_version("cancun")
```

Comment:
maybe we should just mark the whole module -- https://docs.pytest.org/en/latest/example/markers.html#marking-whole-classes-or-modules

Reply:
there is an exception in `test_transient_compiles` tho, should it be moved to a separate file?